website/source/guides/security/vault-pki-integration.html.md

```diff
@@ -0,0 +1,550 @@
+by listing them out:
+
+```
+$ ls /opt/nomad/certs/
+agent.crt  agent.key  ca.crt  cli.crt  cli.key
+```
+
+### Step 11: Configure Nomad to Use TLS
+
+Add the following [tls stanza][nomad-tls-stanza] to your Nomad agent
+configuration file (located at `/etc/nomad.d/nomad.hcl` in this example):
+
+```
+tls {
+  http = true
+  rpc  = true
+
+  ca_file   = "/opt/nomad/certs/ca.crt"
+  cert_file = "/opt/nomad/certs/agent.crt"
+  key_file  = "/opt/nomad/certs/agent.key"
+
+  verify_server_hostname = true
+  verify_https_client    = true
+}
+```
+Send a SIGHUP signal to Nomad to reload the TLS configuration:
+
+```shell
+$ pkill -SIGHUP nomad
+```
+
+If you run `nomad status`, you will now receive the following error:
+
+```
+Error querying jobs: Get http://172.31.52.215:4646/v1/jobs: net/http: HTTP/1.x transport connection broken: malformed HTTP response "\x15\x03\x01\x00\x02\x02"
+```
+
+This is because the Nomad CLI defaults to communicating via HTTP instead of
+HTTPS. We can configure the local Nomad client to connect using TLS and specify
+our custom key and certificates by setting the following environments variables:
+
+```shell
+export NOMAD_ADDR=https://localhost:4646
+export NOMAD_CACERT="/opt/nomad/certs/ca.crt"
+export NOMAD_CLIENT_CERT="/opt/nomad/certs/cli.crt"
+export NOMAD_CLIENT_KEY="/opt/nomad/certs/cli.key"
+```
+
+After these environment variables are correctly configured, the CLI will respond
+as expected:
+
+```shell
+$ nomad status
+No running jobs
+```
+
+## Encrypt Server Gossip
+
+At this point all of Nomad's RPC and HTTP communication is secured with mTLS.
+However, Nomad servers also communicate with a gossip protocol, Serf, that does
+not use TLS:
+
+* HTTP - Used to communicate between CLI and Nomad agents. Secured by mTLS.
+* RPC - Used to communicate between Nomad agents. Secured by mTLS.
+* Serf - Used to communicate between Nomad servers. Secured by a shared key.
+
+Nomad server's gossip protocol use a shared key instead of TLS for encryption.
+This encryption key must be added to every server's configuration using the
+[`encrypt`](/docs/configuration/server.html#encrypt) parameter or with the
+[`-encrypt` command line option](/docs/commands/agent.html).
+
+The Nomad CLI includes a `operator keygen` command for generating a new secure
+gossip encryption key:
+
+```shell
+$ nomad operator keygen
+cg8StVXbQJ0gPvMd9o7yrg==
+```
+
+Alternatively, you can use any method that base64 encodes 16 random bytes:
+
+```shell
+$ openssl rand -base64 16
+raZjciP8vikXng2S5X0m9w==
+$ dd if=/dev/urandom bs=16 count=1 status=none | base64
+LsuYyj93KVfT3pAJPMMCgA==
+```
+
+Put the same generated key into every server's configuration file or command
+line arguments:
+
+```hcl
+server {
+  enabled = true
+
+  # Self-elect, should be 3 or 5 for production
+  bootstrap_expect = 1
+
+  # Encrypt gossip communication
+  encrypt = "cg8StVXbQJ0gPvMd9o7yrg=="
+}
+```
+
```

Comment:
We might need to call out that Gossip encryption is not HUP-able and requires the agent to be restarted.